pkg/operator/controller/canary/controller.go

```diff
@@ -117,9 +167,122 @@ type reconciler struct {
 	client client.Client
 }
 
-// TODO: Canary Controller Phase 2
-// Add callers for these 2 functions
-//
+func (r *reconciler) startCanaryRoutePolling(stop <-chan struct{}) error {
+	// Keep track of how many canary checks have passed
+	// so the route endpoint can be periodically cycled.
+	checkCount := 0
+
+	// Keep track of successive canary check failures
+	// for status reporting.
+	successiveFail := 0
+
+	// Signal that the canary polling loop is running.
+	routeProbeRunning = true
+
+	go wait.Until(func() {
+		// Get the current canary route every iteration in case it has been modified
+		haveRoute, route, err := r.currentCanaryRoute()
+		if err != nil {
+			log.Error(err, "failed to get current canary route for canary check")
+			return
+		} else if !haveRoute {
+			log.Info("canary check route does not exist")
+			return
+		}
+		// Periodically rotate route endpoint after every 5 successful checks.
+		if checkCount == 6 {
```

Comment:
Consider using a variable (e.g. maxChecks)  to store the number of successful checks to compare to checkCount.  If we need to change it later, or decide to allow it to be configurable, there is one variable to change.  Also, comparing checkCount > maxChecks might be best if checkCount can somehow grow larger than maxChecks + 1.

Reply:
Fixed. I actually meant to do this sooner :grin: Thanks!